tket/src/Circuit/macro_circ_info.cpp

```diff
@@ -778,15 +814,28 @@ Circuit::SliceIterator::SliceIterator(
     const Circuit& circ, const std::function<bool(Op_ptr)>& skip_func)
     : cut_(), circ_(&circ) {
   cut_.init();
+  // add qubits to u_frontier
   for (const Qubit& q : circ.all_qubits()) {
     Vertex in = circ.get_in(q);
     cut_.u_frontier->insert({q, circ.get_nth_out_edge(in, 0)});
   }
+
+  // add bits to u_frontier and b_frontier
   for (const Bit& b : circ.all_bits()) {
     Vertex in = circ.get_in(b);
     cut_.b_frontier->insert({b, circ.get_nth_b_out_bundle(in, 0)});
     cut_.u_frontier->insert({b, circ.get_nth_out_edge(in, 0)});
   }
+
+  // add wasmuid to u_frontier
+  if (circ.wasm_added) {
+    Vertex in = circ.get_in(circ.wasmwire);
+    cut_.u_frontier->insert({circ.wasmwire, circ.get_nth_out_edge(in, 0)});
+    // throw std::logic_error("added wasm to u frontier");
```

Comment:
Delete?

Reply:
Removed in [suggested changes I](https://github.com/CQCL/tket/pull/743/commits/e06543a3343027c8ff962d77fda395d3b22d9f96)
[e06543a](https://github.com/CQCL/tket/pull/743/commits/e06543a3343027c8ff962d77fda395d3b22d9f96)